File: services/service-b/src/configs/service.py
import os
import uuid

UUID = uuid.uuid4()
PORT = os.getenv("FASTAPI_PORT", 8000)
HOST = os.getenv("FASTAPI_HOST", "localhost")

#checker_name: str = f"{UUID}-health-check"

class ServiceConfig:
  id: str = f"fastapi-service-{UUID}"
  name: str = "fastapi-service"
  address: str = HOST
  port: int = int(PORT)
  health: str = f"http://{HOST}:{PORT}/health"

File: services/service-b/src/config/service.ts
import { v4 } from "uuid";

const {
  EXPRESS_HOST,
  EXPRESS_PORT,
} = process.env;

const UUID = v4();

const EXPRESS_HOST_PARSED = `http://${EXPRESS_HOST}:${EXPRESS_PORT}`;

const EXPRESS_PORT_PARSED = parseInt(EXPRESS_PORT!, 10);

const CHECKER_NAME = `${UUID}-health-check`;

const CHECKER_HTTP = `http://${EXPRESS_HOST}:${EXPRESS_PORT}/health`;


export const ServiceConfig = {
  id: UUID,
  name: "express-service-b",
  host: EXPRESS_HOST_PARSED,
  port: EXPRESS_PORT_PARSED,
  checker: {
    name: CHECKER_NAME,
    http: CHECKER_HTTP,
  },
};